File: src/plugin/discord/Context.java
package plugin.discord;

import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.entities.User;
import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion;
import net.dv8tion.jda.api.utils.messages.MessageCreateData;
import plugin.utils.Permission;

import java.util.List;

import static plugin.PVars.gamemode;
import static plugin.utils.Permission.getPermsByDiscordId;

public class Context {
    public User author;
    public Message message;
    public MessageChannelUnion channel;
    public List<Message.Attachment> attachments;

    public Context(Message message, MessageChannelUnion channel, User author) {
        this.message = message;
        this.channel = channel;
        this.author = author;
        this.attachments = this.message.getAttachments();
    }

    public boolean hasPerm(Permission perm) {
        boolean has = getPermsByDiscordId(author.getIdLong()).contains(perm);
        if (!has) {
            reply("No access! You need " + perm.name());
        }
        return has;
    }

    public void reply(String content) {
        message.reply(content).queue();
    }

    public void replyServer(String content) {
        message.reply("["+gamemode.simpleName+"] "+content).queue();;
    }

    public void replyEmbed(MessageEmbed embed) {
        message.reply(MessageCreateData.fromEmbeds(embed)).queue();
    }
}

File: src/plugin/discord/Context.kt
package plugin.discord

import net.dv8tion.jda.api.entities.Message
import net.dv8tion.jda.api.entities.Message.Attachment
import net.dv8tion.jda.api.entities.MessageEmbed
import net.dv8tion.jda.api.entities.User
import net.dv8tion.jda.api.entities.channel.unions.MessageChannelUnion
import net.dv8tion.jda.api.utils.messages.MessageCreateData
import plugin.PVars
import plugin.database.models.Permission
import plugin.database.models.Permission.Companion.getPermsByDiscordId

class Context(var message: Message, var channel: MessageChannelUnion, var author: User, var global: Boolean) {
    var attachments: List<Attachment> = this.message.attachments

    fun hasPerm(perm: Permission): Boolean {
        val has: Boolean = getPermsByDiscordId(author.idLong).contains(perm)
        if (!has) {
            reply("No access! You need " + perm.name)
        }
        return has
    }

    fun reply(content: String?) {
        val prefix = if (global) "[Global]" else "[${PVars.gamemode.name}]"
        message.reply("$prefix $content").queue()
    }

    fun replyServer(content: String?) {
        val prefix = if (global) "[Global Server]" else "[${PVars.gamemode.simpleName}]"
        message.reply("$prefix $content").queue()
    }

    fun replyEmbed(embed: MessageEmbed) {
        message.reply(MessageCreateData.fromEmbeds(embed)).queue()
    }
}